### DISCUSSION TITLE
Square Frame

### DISCUSSION THREAD

**Discussion by D4rk-S0ul** [TomSchimansky/CustomTkinter#1016] [Q&A]
Hey, the following picture represents my frame layout. I'm trying to add a square shaped frame (red square). Its center should be located in the center of its master frame and it should react to rescaling the window, always being as big as possible while still being a square.
![image](https://user-images.githubusercontent.com/83767949/211213758-f87c0843-6ebc-4e58-866b-3c8d7e331307.png)
I just can't figure out how to add this square shaped frame, any ideas?

Additionally, I'd be cool to have a little space between the outer line of the red frame its master, how would I do this?

**Comment by Jimmeywho** [comment]
Have you tried setting rows and columns to provide you with the margin?
Using .grid_rowconfigure as an example, setting the weight to 0 would make them inflexible and remain the same thickness as set.

Not tried the square window but maybe limit the resizing to one axis, and then bind a function that, on the resize event, applies the new size to the other axis. This will then update as the user resizes.

Not able to provide examples of what I mean currently but can try and have a look when home later.
